test/ModuleInterface/preserve-type-repr.swift

```diff
@@ -1,5 +1,23 @@
-// RUN: %target-swift-frontend -typecheck -emit-module-interface-path - %s -enable-library-evolution -module-name PreferTypeRepr -module-interface-preserve-types-as-written | %FileCheck %s --check-prefix PREFER
-// RUN: %target-swift-frontend -typecheck -emit-module-interface-path - %s -enable-library-evolution -module-name PreferTypeRepr | %FileCheck %s --check-prefix DONTPREFER
+// RUN: %empty-directory(%t)
+// RUN: %target-swift-frontend %s -typecheck -enable-library-evolution -emit-module-interface-path %t/External.swiftinterface -module-name External -DEXTERNAL
+// RUN: %target-swift-frontend -typecheck -emit-module-interface-path - %s -enable-library-evolution -module-name PreferTypeRepr -module-interface-preserve-types-as-written -I %t | %FileCheck %s --check-prefix PREFER
+// RUN: %target-swift-frontend -typecheck -emit-module-interface-path - %s -enable-library-evolution -module-name PreferTypeRepr -I %t | %FileCheck %s --check-prefix DONTPREFER
+
+#if EXTERNAL
+import External
+
+public struct Toy {
+    public init() {}
+}
+#endif
```

Comment:
This looks a little funky:

1. Do you actually want to `#else` the rest of the test and then `#endif` at the end?
2. How are you `import External`-ing inside the module interface for `External`? Is this an overlay?

Reply:
Fixed. The `import External` seems to somehow work even when the module is not an overlay. 🤔